File: java/src/beginner/FizzBuzz.java
package beginner;

import java.util.Arrays;

public class FizzBuzz {
  public static String mapIntToFizzBuzz(int i) {
    if (i % 3 == 0 && i % 5 == 0) {
      return "FizzBuzz";
    } else if (i % 3 == 0) {
      return "Fizz";
    } else if (i % 5 == 0) {
      return "Buzz";
    } else {
      return Integer.toString(i);
    }
  }

  public static String[] fizzBuzz(int n) {
    String[] output = new String[n];
    for (int i = 1; i <= n; i++) {
      output[i - 1] = mapIntToFizzBuzz(i);
    }
    return output;
  }

  public static void main(String[] args) {
    System.out.println("Testing fizz buzz");
    assert (Arrays.equals(fizzBuzz(3), new String[] { "1", "2", "Fizz" }));
    System.out.println("All tests passed!");
  }
}

File: js/src/beginner/fizzBuzz.js
import { strict as assert } from 'node:assert'
import logger from '../utils.js'

function fizzBuzz(n) {
  const mapFizzBuzz = (i) => {
    switch (true) {
      case i % 3 === 0 && i % 5 === 0:
        return 'FizzBuzz'
      case i % 3 === 0:
        return 'Fizz'
      case i % 5 === 0:
        return 'Buzz'
      default:
        return String(i)
    }
  }
  return Array(n)
    .fill(null)
    .map((_, i) => mapFizzBuzz(i + 1))
}

export function test() {
  logger.log('Running test for fizz buzz')
  assert.deepEqual(fizzBuzz(3), ['1', '2', 'Fizz'])
  logger.log('All test passed!')
}
